src/finch/tensor.py

```diff
@@ -16,17 +15,36 @@ def __str__(self):
 
 
 class Tensor(_Display):
-    def __init__(self, lvl=None, arr=None, jl_data=None):
+    """
+    A wrapper class for Finch.Tensor and Finch.SwizzleArray
+    """
+    ROW_MAJOR = "C"
+    COLUMN_MAJOR = "F"
+
+    def __init__(self, lvl=None, arr=None, jl_data=None, order=None):
         if arr is not None and not isinstance(arr, np.ndarray):
-            raise ValueError("For now only numpy input allowed")
+            raise ValueError("For now only numpy input allowed.")
+        if order not in (None, self.ROW_MAJOR, self.COLUMN_MAJOR):
+            raise ValueError(f"order must be a 'C' or 'F', but is: {type(order)}.")
 
+        # constructor for levels description and NumPy array input
         if lvl is not None and arr is not None and jl_data is None:
-            self._obj = jl.Tensor(lvl._obj, np.array(arr, order="F"))
-        elif jl_data is not None:
+            if order == self.ROW_MAJOR or order is None:
+                permutation = tuple(range(1, arr.ndim + 1)[::-1])
+                self.order = self.ROW_MAJOR
+            elif order == self.COLUMN_MAJOR:
+                permutation = tuple(range(1, arr.ndim + 1))
+                self.order = order
```

Comment:
why do we maintain this in the python Tensor class as well as the julia swizzle class? I think it would be easier to maintain one source of truth for the order.

Reply:
I think that we need to maintain `order` in the python Tensor and julia swizzle class.

`SwizzleArray` defines **current** permutation, where `order` in the python Tensor **original** permutation.

Let's assume a user creates a tensor that is row-major, so the permutation is `[n, n-1, ..., 1]`. This permutation is set in the `order` attribute and `SwizzleArray` internal object. Then the user performs and arbitrary permutation `p`. This permutation is set to the internal `SwizzleArray`.
Then the user calls `todense()`. We need to know how to move to the NumPy's row-major order. We can't just use what is in `SwizzleArray` as it has user's permutation. We use original `order` to know how to move to the actual row-major after materializing the internal `SwizzleArray`.

1. when created: order: (3, 2, 1), swizzle array: (3, 2, 1)
2. custom user permutation: order (3, 2, 1), swizzle array (2, 1, 3)
3. calling `todense()`: order (3, 2, 1) <-- we still need it to move back to NumPy!, swizzle array (2, 1, 3) <-- just used for materialization.

Reply:
Ah, I see what you're going for here. The idea is that the "original" order of the tensor is what the user would want when they call `todense`. I disagree on this point though, because the user could also create a dense finch tensor and I think we want to have the property that todense() is non-copying whenever possible. I think we want to build a system where the following is always true, right? (ignoring syntax errors):

```
tns = Tensor(Dense(Dense(Dense(Element(0.0))), order=order)
tns_2 = permutedims(tns, some_other_order)
tns_2.strides == tns_2.todense().strides
```

To do this, I think we need only set `order` in todense to be [3, 2, 1]